### PR TITLE
Dynamic form component + directive

### PR DESCRIPTION
**Pull request by FarajiA**
## JIRA:
See ticket here - https://jira.rax.io/browse/MNRVA-436

 ## Description:
For the purpose of creating & updating a monitor this will allow form fields to dynamically generate fields based on the Salus (Monitoring) API. 

As support for more monitors are added to the app more types of fields will have to be  This dynamic component so far supports the following types of fields:

- Input
- Checkbox
- Select

 ## Testing:
`npm run test` - 

 ## Screenshots:
(if applicable)

### CODE REVIEW

File: src/app/_features/monitors/components/input/input.component.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  selector: 'monitor-input',
+  templateUrl: './input.component.html',
+  styleUrls: ['./input.component.scss']
+})
+export class InputComponent implements OnInit {
+
+  config: FieldConfig;
+  group: FormGroup;
+
+  @ViewChild('content', {read: TemplateRef}) actualContentTmpl: TemplateRef<any>;
```
Comment (pras7430) [inline src/app/_features/monitors/components/input/input.component.ts:15]: Hey, I have noticed during the version update if wanted to use query result in ngoninit then it should be {static:true}.

see it:
https://angular.io/guide/static-query-migration

File: src/app/_features/monitors/components/select/select.component.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  selector: 'monitor-select',
+  templateUrl: './select.component.html',
+  styleUrls: ['./select.component.scss']
+})
+export class SelectComponent implements OnInit {
+
+  config: FieldConfig;
+  group: FormGroup;
+
+  @ViewChild('content', {read: TemplateRef}) actualContentTmpl: TemplateRef<any>;
```
Comment (pras7430) [inline src/app/_features/monitors/components/select/select.component.ts:15]: If you update above then we need to update here as well

File: src/app/_features/monitors/mon.utils.ts [hunk 3]
```diff
@@ -37,6 +40,107 @@ function FormatMonitorUtil(type: string, formData: any): CreateMonitor {
...
+    }
+    return fields;
+}
+
+/**
+ *
+ * @param monitor a monitor based on schema type
+ * @param fieldName name of field of monitor.properties
+ */
+function createField(monitor, fieldName): FieldConfig  {
```
Comment (pras7430) [inline src/app/_features/monitors/mon.utils.ts:65]: hey, this function complexity is very high, what if we divide it in smaller functions?

File: src/app/_features/monitors/components/input/input.component.ts (hunk 1 above)
Comment (FarajiA) [reply src/app/_features/monitors/components/input/input.component.ts]: Yup, this is been updated

File: src/app/_features/monitors/mon.utils.ts (hunk 3 above)
Comment (FarajiA) [reply src/app/_features/monitors/mon.utils.ts:40]: I may you need to show me what you mean, though I do get your point this function does have a lot going on thought using the reveal pattern would be the simplest approach.

**Pull request merged**